sharding/nodesCoordinator/indexHashedNodesCoordinator_test.go

```diff
@@ -2519,3 +2519,97 @@ func TestIndexHashedNodesCoordinator_GetShardValidatorInfoData(t *testing.T) {
 		require.Equal(t, svi, shardValidatorInfo)
 	})
 }
+
+func TestNodesCoordinator_CustomConsensusGroupSize(t *testing.T) {
+	arguments := createArguments()
+	eligibleMap := createDummyNodesMap(3, 2, "eligible")
+	waitingMap := createDummyNodesMap(0, 2, "waiting")
+	arguments.EligibleNodes = eligibleMap
+	arguments.WaitingNodes = waitingMap
+	arguments.ValidatorInfoCacher = dataPool.NewCurrentEpochValidatorInfoPool()
+	arguments.ChainParametersHandler = &shardingmock.ChainParametersHandlerStub{
+		ChainParametersForEpochCalled: func(epoch uint32) (config.ChainParametersByEpochConfig, error) {
+			if epoch < 3 {
+				return config.ChainParametersByEpochConfig{
+					ShardConsensusGroupSize:     2,
+					ShardMinNumNodes:            2,
+					MetachainConsensusGroupSize: 3,
+					MetachainMinNumNodes:        3,
+				}, nil
+			}
+
+			if epoch < 6 {
+				return config.ChainParametersByEpochConfig{
+					ShardConsensusGroupSize:     3,
+					ShardMinNumNodes:            3,
+					MetachainConsensusGroupSize: 3,
+					MetachainMinNumNodes:        3,
+				}, nil
+			}
+
+			if epoch < 9 {
+				return config.ChainParametersByEpochConfig{
+					ShardConsensusGroupSize:     3,
+					ShardMinNumNodes:            3,
+					MetachainConsensusGroupSize: 2,
+					MetachainMinNumNodes:        2,
+				}, nil
+			}
+
+			return config.ChainParametersByEpochConfig{
+				ShardConsensusGroupSize:     3,
+				ShardMinNumNodes:            3,
+				MetachainConsensusGroupSize: 3,
+				MetachainMinNumNodes:        3,
+			}, nil
+		},
+	}
+
+	ihnc, _ := NewIndexHashedNodesCoordinator(arguments)
+	require.NotNil(t, ihnc)
+
+	header := &block.MetaBlock{
+		PrevRandSeed: []byte("rand seed"),
+		EpochStart:   block.EpochStart{LastFinalizedHeaders: []block.EpochStartShardData{{}}},
+	}
+
+	// change to epoch 1 - should have 3 eligible per shard, 3 per meta
```

Comment:
I would also add tests for epochs 2, 3, 4, 6, 8, 9, 10, and 11 to test the correct change conditions

Reply:
refactored the unit test and now it checks from epoch 0 to epoch 100